core/src/test/java/org/elasticsearch/test/ESSingleNodeTestCase.java

```diff
@@ -118,16 +119,18 @@ protected boolean resetNodeAfterTest() {
 
     private static Node newNode() {
         Node build = NodeBuilder.nodeBuilder().local(true).data(true).settings(Settings.builder()
-            .put(ClusterName.SETTING, InternalTestCluster.clusterName("single-node-cluster", randomLong()))
-            .put("path.home", createTempDir())
-            .put("node.name", nodeName())
-            .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1)
-            .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 0)
-            .put("script.inline", "on")
-            .put("script.indexed", "on")
-            .put(EsExecutors.PROCESSORS, 1) // limit the number of threads created
-            .put("http.enabled", false)
-            .put(InternalSettingsPreparer.IGNORE_SYSTEM_PROPERTIES_SETTING, true) // make sure we get what we set :)
+                .put(ClusterName.SETTING, InternalTestCluster.clusterName("single-node-cluster", randomLong()))
+                .put("path.home", createTempDir())
+                .put("path.shared_data", createTempDir().getParent())
```

Comment:
I would like it better if custom data paths were explicitely created in a specific drectory. However it's an existing issue that your PR does not introduce, so could you just add a comment explaining why you set the custom data path this way and add a TODO to explicitely set custom index paths to be sub directories?